components/setup/DockerInstructions.tsx

```diff
@@ -15,230 +11,61 @@ interface DockerInstructionsProps {
-		axios.post('/api/dpu/pubsub', { userId }).then(async (response) => {
-			if (response.data.installId) {
-				setInstallId(response.data.installId);
-				if (selectedInstallationType === 'individual' && selectedProvider === 'github') {
-					if (!session) {
-						console.error(`[DockerInstructions] could not get session for userId: ${userId}`);
-						return;
-					}
-					if (isRepoSelectionDone) {
-						setSelfHostingCode(`
+const DockerInstructions: React.FC<DockerInstructionsProps> = ({ userId, selectedInstallationType, selectedProvider, session, installId }) => {
+	let selfHostingCode;
+	if (selectedInstallationType === 'individual' && selectedProvider === 'github') {
+		if (!session) {
+			console.error(`[DockerInstructions] could not get session for userId: ${userId}`);
+			selfHostingCode = `Unable to get topic name for user\nPlease refresh this page and try again.`;
+		} else {
+			selfHostingCode = `
 docker pull asia.gcr.io/vibi-prod/dpu/dpu &&\n
-docker run -e INSTALL_ID=${response.data.installId} \\
+docker run -e INSTALL_ID=${installId} \\
 -e PROVIDER=<your_provider_here> \\
 -e GITHUB_PAT=<Your gh cli token> \\
 asia.gcr.io/vibi-prod/dpu/dpu
-					`);
-					}
-					else {
-						let providerReposForUser = await getUserReposForProvider(session, 'github');
-						setAllRepos(providerReposForUser);
-					}
-				} else if (selectedInstallationType === 'individual' && selectedProvider === 'bitbucket') {
-					setSelfHostingCode(`
-Coming Soon!
-					`)
-				} else if (selectedInstallationType === 'project' && selectedProvider === 'bitbucket') {
-					if (!session) {
-						console.error(`[DockerInstructions] could not get session for userId: ${userId}`);
-						return;
-					}
-					if (isRepoSelectionDone) {
-						setSelfHostingCode(`
-docker pull asia.gcr.io/vibi-prod/dpu/dpu &&\n
-mkdir -p ~/.config/vibinex &&\n
-docker run -e INSTALL_ID=${response.data.installId} -v ~/.config/vibinex:/app/config asia.gcr.io/vibi-prod/dpu/dpu
-					`);
-					}
-					else {
-						let providerReposForUser = await getUserReposForProvider(session, 'bitbucket');
-						setAllRepos(providerReposForUser);
-					}
-				} else {
-					setSelfHostingCode(`
-docker pull asia.gcr.io/vibi-prod/dpu/dpu &&\n
-mkdir -p ~/.config/vibinex &&\n
-docker run -e INSTALL_ID=${response.data.installId} -v ~/.config/vibinex:/app/config asia.gcr.io/vibi-prod/dpu/dpu
-					`);
-				}
-			}
-		}).catch((error) => {
-			setSelfHostingCode(`Unable to get topic name for user\nPlease refresh this page and try again.`);
-			console.error(`[DockerInstructions] Unable to get topic name for user ${userId} - ${error.message}`);
-		});
-	}, [userId, selectedInstallationType, selectedProvider]);
-
-	const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>, repo: RepoIdentifier) => {
-		if (event.target.checked) {
-			setSelectedRepos([...selectedRepos, repo]);
-		} else {
-			setSelectedRepos(selectedRepos.filter(selectedRepo => selectedRepo !== repo));
+					`;
```

Comment:
This should also have volume mount option for saving auth

Reply:
Will we implement this for individual installation also?